src/shogun/kernel/string/WeightedDegreeStringKernel.cpp

```diff
@@ -565,12 +560,11 @@ bool WeightedDegreeStringKernel::set_wd_weights_by_type(EWDKernType p_type)
 	ASSERT(degree>0)
 	ASSERT(p_type==E_WD) /// if we know a better weighting later on do a switch
 
-	SG_FREE(weights);
-	weights=SG_MALLOC(float64_t, degree);
+	weights=SGMatrix<float64_t>(degree, 1);
 	weights_degree=degree;
 	weights_length=1;
 
-	if (weights)
+	if (weights.size() > 0)
```

Comment:
just wondering whether these files are covered by any sort of tests ... as bugs might slip through otherwise? (Even if unlikely)

Reply:
there are no unittests.. It might be worth adding something that is run before these changes

Reply:
think integration tests are fine for that... but then if we port the string kernel ones you will get many conflicts in here....any idea how to best do this? we could involve the gsoc students maybe?

Reply:
Hmm I think the easiest is if I just run the code in the develop branch and copy paste the results into a unittest?

Reply:
The issue is that the interface cannot be accessed from outside? Oh but we don't need it I guess. Yes, @LiuYuHui or @Hephaestus12 could write a meta example for this then. Shouldn't be difficult to rebase

Reply:
I personally think unit tests should be hand-designed to cover corner cases or reference libraries and not be created from outputs, this is what our integration meta tests are for

Reply:
> I personally think unit tests should be hand-designed to cover corner cases or reference libraries and not be created from outputs, this is what our integration meta tests are for

yes agreed. For some reason I thought we couldn't write a meta example with this kernel. But we can right? Just cannot use factories

Reply:
Yeah. The GSoC students could prioritize the meta examples to port being the string kernel ones. And then they have to work with what API is there (no factories yet). @LiuYuHui @Hephaestus12 let us know if you have any questions, it would be great if you both could port a few string kernel examples next

Reply:
the examples that would be highest priority would be `kernel_weighted_degree_string.py` and `kernel_weighted_degree_position_string.py`. Those are the ones I had to refactor quite a bit

Reply:
how to handle this? https://github.com/shogun-toolbox/shogun/blob/55bb4a991f9336c78225545c9ddb530ca31e73cc/examples/undocumented/python/kernel_weighted_degree_string.py#L19-L22

Reply:
you could write the array manually, have a look at the matrix_types meta example

Reply:
How do you replace this line?
just `create_features()` doesn't seem to work.

https://github.com/shogun-toolbox/shogun/blob/41b6247f0a8c0f9ce45b388ae085d42d90bdd16d/examples/undocumented/python/kernel_linear_string.py#L14-L15

Reply:
There is a create_string_features function with that same signature